### DISCUSSION TITLE
Anyone tried using the API to create a iOS Shortcut to save links from iPhone?

### DISCUSSION THREAD

**Discussion by relink2013** [goniszewski/grimoire#140] [Q&A]
I see theres no mobile apps on the roadmap, I really want to give this a go but I have to be able to add links from my phone. 

But if Grimoire has an API then it more than likely possible to create a shortcut for iOS. I just wouldn't know how to actually make it work.

**Comment by goniszewski** [comment]
Hello @relink2013! Firstly, sorry for not getting back to you sooner.

I have looked into this topic and came with a quick Apple Shortcut template you can configure yourself and use: https://www.icloud.com/shortcuts/4d84bb8c7872429aa2d486dc66a4164e

It requires you to set:
- correct URL to your Grimoire instance
- generating a valid token for the authorization header (for now, you need to send POST request to `<GRIMOIRE_URL>/api/auth` with a JSON body containing an object with `login` (username or e-mail) and `password`
- pasting this token into the last step's (_Get contents of **Text**_) Authorization header in the place of `key` (just after `Bearer `

As for generating the token, I plan to add the ability to specify your own long-term API tokens for such integrations in the user's settings page. The existing one is being used by the web extension only.

**Comment by relink2013** [comment]
Awesome, added it to my shortcuts and I'll be trying it soon.

**Comment by singhdilveer1** [comment]
Heres’s something I cooked up for getting bearer tokens from grimoire, the token will be copied to the clipboard.  -
Grimoire Tokens :
https://www.icloud.com/shortcuts/4a500a03bfe243c785ea0af7dc0c87f1

A quick modification I did because the “Save to Grimoire” shortcut by [goniszewski](https://github.com/goniszewski) didn’t quite work -

Grimoire :
https://www.icloud.com/shortcuts/0293d753297040999d880b2c8656e7bb

1. Install the first shortcut, run it and the bearer token will be copied to the clipboard.
2. Install the second shortcut and paste it in the bearer token field during setup. And you’re all set.
